src/cctp-sdk/index.ts

```diff
@@ -32,112 +32,129 @@ interface AttestationStatus {
   attestation: string;
 }
 
+interface Authorization {
+  validAfter: number;
+  validBefore: number;
+  nonce: Uint8Array;
+  v: number;
+  r: string;
+  s: string;
+}
+
 export const transfer = async (
   amount: bigint,
   srcMaxFee: bigint,
   dstMaxFee: bigint,
-  dstChainId: number,
+  dstChainId: bigint,
   signer: ethers.Wallet
 ): Promise<void> => {
   if (srcMaxFee + dstMaxFee > amount)
     throw new Error("Max fee amount exceeds total amount");
 
-  const srcChainId = await signer.getChainId();
+  const provider = signer.provider;
+  if (!provider) throw new Error("Signer missing provider");
 
+  const { chainId: srcChainId } = await provider.getNetwork();
   if (srcChainId === dstChainId)
     throw new Error("Source and destination chain must be different");
 
-  const srcChain = CHAIN[srcChainId];
-  const dstChain = CHAIN[dstChainId];
+  const srcChain = CHAIN[srcChainId.toString()];
+  const dstChain = CHAIN[dstChainId.toString()];
 
   if (!srcChain || !dstChain) throw new Error("Unsupported chain");
 
-  const depositAuthorization = await buildReceiveWithAuthorization(
+  const srcAuthorization = await buildAuthorization(
     signer,
     srcChain.usdc,
     amount,
-    srcChain.forwarder,
-    srcChainId,
-    "0x9b94f9a1"
+    srcChain.sender,
+    srcChainId
   );
 
-  const withdrawAuthorization = await buildReceiveWithAuthorization(
+  const dstAuthorization = await buildAuthorization(
     signer,
     dstChain.usdc,
     dstMaxFee,
-    dstChain.forwarder,
-    dstChainId,
-    "0xd0f48715"
+    dstChain.receiver,
+    dstChainId
   );
 
   const relay = new GelatoRelay();
-  const forwarder = new ethers.utils.Interface(ForwarderAbi);
 
-  const deposit = forwarder.encodeFunctionData("deposit", [
+  const sender = new ethers.Interface(SenderAbi);
+  const receiver = new ethers.Interface(ReceiverAbi);
+
+  const depositForBurn = sender.encodeFunctionData("depositForBurn", [
+    amount,
     srcMaxFee,
     dstChain.domain,
-    depositAuthorization,
+    srcAuthorization,
   ]);
 
   const tx = await relayAndWait(
```

Comment:
nice to first finish Relay params - so first have another signature step and then `relayAndWait`